include/core/JamomaSample.h

```diff
@@ -244,7 +254,92 @@ namespace Jamoma {
 			for (int i=0; i<frameCount; ++i)
 				values[i] = vector[i];
 		}
-		
+        
+        
+        /** Duplicate values at the beginning and end to provide padding for interpolation.
+            @param  paddingAmount   Number of frames to add to both beginning & end of SampleBundle. Total number of frames added to SampleBundle will be twice this amount.
+         */
+        void applySamplePadding(int paddingAmount)
+        {
+            
+            // ensure that we are adding at least one sample, AND not more than half SampleBundle frames
+            assert(paddingAmount > 0 && paddingAmount < mFrameCount/2);
+            
+            // make a copy
+            SampleBundleData newPaddedChannels = mChannels;
+            
+            for (auto& channel : newPaddedChannels) {
+                
+                // set up temporary vectors for padding samples
+                SampleVector firstSamples(paddingAmount,0.0);
+                SampleVector lastSamples(paddingAmount,0.0);
+                
```

Comment:
Question for you @nwolek : We appear to do the following (note that I use source and dest but I know they are the same):

0. Have a source SampleVector
1. Create a temp SampleVector
2. Assign values (init to zero) the temp SampleVector
3. Create another temp SampleVector
4. Assign values (init to zero) the second temp SampleVector
5. Copy samples from source to the temp vector
6. Copy samples from source to the second temp vector
7. Copy samples from temp to dest (insert call)
8. Copy samples from temp2 to dest (insert call)

Can we eliminate steps 1-6?  It would be nice if we could but I could understand that maybe calling insert would corrupt the iterators or something if we just used iterators from the source passed to the insert.  But I'm sure that it either does something smart or there is an idiom for dealing with this because it can't be that rare.  Not sure if you found anything (even the dead-ends might be good to know about?)